cli/compiler.rs

```diff
@@ -133,6 +142,43 @@ impl ModuleMetaData {
   }
 }
 
+fn new_cmd_id() -> CmdId {
+  let next_rid = C_NEXT_CMD_ID.fetch_add(1, Ordering::SeqCst);
+  next_rid as CmdId
+}
+
+/// Creates a future that will route worker responses
+/// to their respective channels.
+fn handle_worker_message_stream(
+  rid: ResourceId,
+) -> impl Future<Item = (), Error = ()> {
+  let worker_stream = resources::get_message_stream_from_worker(rid);
+  worker_stream
+    .for_each(|msg: Buf| {
+      // All worker responses are handled here first before being sent via
+      // their respective sender. This system can be compared to the
+      // promise system used on the js side. This provides a way to
+      // resolve many futures via the same channel.
+      let res_json = std::str::from_utf8(&msg).unwrap();
+      debug!("Got message from worker: {}", res_json);
+      // Get the intended receiver's cmd_id from the message.
+      let cmd_id = match serde_json::from_str::<serde_json::Value>(res_json) {
```

Comment:
I meant just the parsing part - seems like a good function.

Reply:
for cmd_id?